src/Telegram.Bot/TelegramBotClient.cs

```diff
@@ -338,6 +334,12 @@ public TelegramBotClient(string token, IWebProxy webProxy)
                 {
                     OnReceiveGeneralError?.Invoke(this, generalException);
                 }
+
+                foreach (var update in updates)
+                {
+                    MessageOffset = update.Id + 1;
```

Comment:
I think update offset increment should be after the event handlers invocation. So if something went wrong and exception was thrown the update is not lost when StartReceiving called again on the same bot client instance.